sale_order_line_date/models/sale_order.py

```diff
@@ -15,27 +15,15 @@ class SaleOrder(models.Model):
     @api.onchange("expected_date")
     def _onchange_expected_date(self):
         """
-        Every time a sale order line Delivery Date is changed,
-        this method is called.
+        Override method because it destroys the correct operation.
         """
-        res = super(SaleOrder, self)._onchange_expected_date()
-        dates_list = []
-        for line in self.order_line.filtered(
-            lambda x: x.state != "cancel"
-            and not x._is_delivery()
-            and not x.display_type
-        ):
-            if line.commitment_date:
```

Comment:
Odoo fixed this themselves :tada: https://github.com/odoo/odoo/commit/449bf686418425a6e4d6c0df68c09f6d77177b02

You can remove it completly